src/Compilers/Core/Portable/InternalUtilities/CharMemoryEqualityComparer.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Roslyn.Utilities
+{
+    /// <summary>
+    /// Provide structural equality for ReadOnlyMemory{char} instances.
+    /// </summary>
+    internal sealed class CharMemoryEqualityComparer : IEqualityComparer<ReadOnlyMemory<char>>
+    {
+        public static readonly CharMemoryEqualityComparer Instance = new CharMemoryEqualityComparer();
+
+        private CharMemoryEqualityComparer() { }
+
+        public bool Equals(ReadOnlyMemory<char> x, ReadOnlyMemory<char> y)
+        {
+            if (x.Length != y.Length)
+            {
+                return false;
+            }
+
+            var xSpan = x.Span;
```

Comment:
This should all just be `x.Span.SequenceEquals(y.Span)`. That implementation of `SequenceEquals` is heavily optimized, particularly on CoreCLR.

Reply:
Nice, I didn't know that existed.